R/ranges-add-distance-utils.R

```diff
@@ -0,0 +1,97 @@
+#' Add suffix to column name
+#' 
+#' I think dplyr::add_count will change how they add names in 1.0.0
+#' used to it would add n until it got a unique name
+#' I can't find the blogpost describing the new behavior,
+#' but it may be good to mirror whatever that is here.
+#' For now, just recursively add suffix until a unique column is found
+#'
+#' @param name column name to add
+#' @param suffix suffix to use
+#' @param names names of data.frame in which to add new column
+#'
+#' @return a unique name generated by prepending suffix recursively until
+#'     a unique name is found
+#' @noRd
+add_colname_suffix <- function(name, suffix, names){
+  
+  stopifnot(length(suffix) == 1)
+  
+  if (name %in% names) {
+    name <- paste0(name, suffix)
+  } else {
+    return(name)
+  }
+  add_colname_suffix(name, suffix, names)
+}
+
+#' @param ranges ranges object from join_nearest expand_hits internal function
+#'
+#' @param hits  hits from distanceToNearest
+#' @param colname name of column to hold distance values
+#' @param suffix suffix to add to colname if exists
+#'
+#' @noRd
+add_distance_col <- function(ranges, hits, colname = "distance", suffix = ".y"){
```

Comment:
I'm not sure if this needs to be a method. I think you could  refactor this to use some of the internals I wrote for the overlap left join.

I think the internals could just be:
```suggestion
add_distance_col <- function(ranges, hits, colname) {
  has_hit <-  rep(FALSE, queryLength(hits))
  has_hit[queryHits(hits)] <- TRUE
  mcols(ranges)[[colname]] <-  NA_integer # double brackets protects against the case where mcols(iranges) is NULL i think
  mcols(ranges)[[colname]][has_hit] <- mcols(hits)$distance
}

```

Reply:
This is a good suggestion, and the general strategy works much better.

Unfortunately, the double bracket solution did not work for me, so I fix the issue with an if/else block instead of method dispatch, since `mcols` for IRanges are `NULL` if empty, but are a `DataFrame` with no columns in GRanges. I moved the logic from the IRanges method to this block.